PHP/PHP Source.sublime-syntax

```diff
@@ -208,65 +233,102 @@ contexts:
           pop: true
         - include: expressions
-    - match: (?i)\s*\(\s*(array|real|double|float|int(eger)?|bool(ean)?|string|object|binary|unset)\s*\)
-      captures:
-        1: storage.type.php
-    - match: (?i)\b(array|real|double|float|int(eger)?|bool(ean)?|string|class|clone|function|interface|object)\b
-      scope: storage.type.php
-    - match: (?i)\b(parent|self)\b
-      scope: variable.language.php
-    - match: (?i)\b(global|abstract|const|extends|implements|final|p(r(ivate|otected)|ublic)|static)\b
-      scope: storage.modifier.php
-    - include: object
-    - match: ;
-      scope: punctuation.terminator.expression.php
-    - include: heredoc
-    - include: numbers
-    - match: (\+|-|\*|/|%|&|\||\^|>>|<<|\.|\?\?)=
-      scope: keyword.operator.assignment.augmented.php
-    - match: "=>"
-      scope: keyword.operator.key.php
-    - match: '(?:(\=)(&))|(&(?=[[:alpha:]$_]))'
-      captures:
-        1: keyword.operator.assignment.php
-        2: storage.modifier.reference.php
-        3: storage.modifier.reference.php
-    - match: (@)
-      scope: keyword.operator.error-control.php
-    - match: (\-\-|\+\+)
-      scope: keyword.operator.increment-decrement.php
-    - match: (\-|\+|\*|/|%)
-      scope: keyword.operator.arithmetic.php
-    - match: '&&|\|\|'
-      scope: keyword.operator.logical.php
-    - match: '<<|>>|~|\^|&|\|'
-      scope: keyword.operator.bitwise.php
-    - match: (===|!==|<=>|==|!=|<=|>=|<>|<|>)
-      scope: keyword.operator.comparison.php
-    - match: (?i)(!)|\b(and|or|xor|as)\b
-      scope: keyword.operator.logical.php
-    - include: function-call
-    - match: \.\.\.
-      scope: keyword.operator.spread.php
-    - match: \.
-      scope: keyword.operator.string.php
-    - match: "="
-      scope: keyword.operator.assignment.php
-    - match: '(?i)\b(instanceof)\s+(?=[[:alpha:]\\$_])'
-      captures:
-        1: keyword.operator.type.php
-      push:
-        - match: '(?=[^[:alnum:]_\\$])'
-          pop: true
-        - include: class-name
-        - include: variables
-    - include: instantiation
-    - match: '(?i)(goto)\s+({{identifier}})'
-      captures:
-        1: keyword.control.goto.php
-        2: support.other.php
-    - include: string-backtick
-    - match: '(\[)\s*(\])'
+        - match: (?i)\s*\(\s*(array|real|double|float|int(eger)?|bool(ean)?|string|object|binary|unset)\s*\)
+          captures:
+            1: storage.type.php
+        - match: |-
+            (?ix)\b(?:
+              array
+              | bool
+              | boolean
+              | class
+              | clone
+              | double
+              | float
+              | function
+              | int
+              | integer
+              | interface
+              | object
+              | real
+              | string
+            )\b
+          scope: storage.type.php
+        - match: |-
+            (?ix)\b(?:
+              parent
+              | self
+            )\b
```

Comment:
Personally, I am not a fan about expanding short regexes like this. I think most of problems come from compact regexes for constants and function names.

Reply:
I agree, I'll sort this out when I come back to it.